torch_xla/csrc/elementwise.cpp

```diff
@@ -92,6 +92,33 @@ xla::XlaOp BuildHardSigmoidBackward(xla::XlaOp grad_output, xla::XlaOp input) {
   return xla::Select(Between(input, -3.0, 3.0), grad_output / six, zero);
 }
 
+xla::XlaOp BuildHardSwish(xla::XlaOp input) {
+  const xla::Shape& shape = XlaHelpers::ShapeOfXlaOp(input);
+  xla::XlaOp zero = xla::Zero(input.builder(), shape.element_type());
+  xla::XlaOp three = XlaHelpers::ScalarValue<float>(3.0, shape.element_type(),
+                                                    input.builder());
+  xla::XlaOp six = XlaHelpers::ScalarValue<float>(6.0, shape.element_type(),
+                                                  input.builder());
+  return xla::Mul(input, (xla::Min(xla::Max(input + three, zero), six) / six));
+}
+
+xla::XlaOp BuildHardSwishBackward(xla::XlaOp grad_output, xla::XlaOp input) {
+  const xla::Shape& shape = XlaHelpers::ShapeOfXlaOp(input);
+  xla::XlaOp three = XlaHelpers::ScalarValue<float>(3.0, shape.element_type(),
+                                                    input.builder());
+  xla::XlaOp zero = xla::Zero(input.builder(), shape.element_type());
+  xla::XlaOp one = XlaHelpers::ScalarValue<float>(1.0, shape.element_type(),
+                                                  input.builder());
+  xla::XlaOp pointfive = XlaHelpers::ScalarValue<float>(
+      0.5, shape.element_type(), input.builder());
+
+  xla::XlaOp stepone =
+      xla::Select(Between(input, -3.0, 3.0),
+                  xla::Mul(grad_output, pointfive + (input / three)), zero);
+
+  return xla::Select(xla::Ge(input, three), one, stepone);
```

Comment:
The derivative for `x>=3` seems to be `1 * grad_output`; right?

Looks like this line should be:

```
return xla::Select(xla::Ge(input, three), grad_output, stepone);
```

Then, we won't need these [lines](https://github.com/pytorch/xla/pull/3496/files#diff-777be0b41499d13e65c42f07f961ee5810a4cd79e66665fa8da1519edcd8b31aR110-R111).

Reply:
Great spot! Thanks for double checking this.